### PR TITLE
Improve native rocksdb dependencies loading

### LINKED ISSUE #4039: On AMD Ubuntu 22.04, can't run nethermind due to missing RocksDB
**Describe the bug**
I installed nethermind v1.13.0 on Ubuntu 22.04 on an Intel system according to the PPA instructions at https://github.com/NethermindEth/nethermind/releases.

When I try to run it using those instructions, I get a long set of errors, beginning with "2022-05-19 08:20:44.7240|RocksDb was not found, please make sure it is installed on your machine."

I tried installing RocksDB using a couple of different methods, but they haven't helped. Apparently I haven't installed it in a way that nethermind can connect to, and I don't know what to do about it.

It seems like, if RocksDB is required as a prerequisite, the instructions should say so, and it would be very helpful if you  say how to install it in a way that makes it useful for nethermind. 

**To Reproduce**
On Ubuntu 22.04 on an Intel machine:

sudo add-apt-repository ppa:nethermindeth/nethermind
sudo apt install nethermind
nethermind --config mainnet

**Expected behavior**
It works without exiting due to error

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - Browser [e.g. chrome, safari]: None used

**Additional context**
I also tried installing nethermind by downloading the zip, using apt to install dependencies, etc. Those instructions are also listed on https://github.com/NethermindEth/nethermind/release. I get the same results as above. I notice that the instructions to install nethermind for Arm64 include installing librocksdb5.17 as a dependency. However, that library doesn't appear to exist for Amd64 Ubuntu and it isn't mentioned in the non-Arm64 Linux instructions on that page.

### PR DESCRIPTION
**Pull request by flcl42**
Fixes #4039

If Nethermind is downloaded from sources, it's easy to forget to make a library link that fixes RocksDb native dependency loading

## Changes:
- Load libdl.so.2 if libdl.so was not found

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

## Manual tests

- The client runs and shutdowns without problems on fresh linux installation after being downloaded as an archive/installed as a ppa package and in any other case too

### CODE REVIEW

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
just make it static in static constructor.

**Review by FalcoXYZ — APPROVED** [review]
LGTM! 
Tested on Ubuntu 22 arm64, works without the symlink command.

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 1]
```diff
@@ -66,6 +68,15 @@ protected static void InitCache(IDbConfig dbConfig)
         }
     }
 
+    private static readonly Dictionary<string, string> MissingNativeLibraryNameMapping = new() { { "libdl", "libdl.so.2" } };
+
+    private static IntPtr OnResolvingUnmanagedDll(Assembly _, string nativeLibraryName) =>
+        MissingNativeLibraryNameMapping.ContainsKey(nativeLibraryName)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:74]: btw its just one, why not hardcode instead of dictionary?

Comment (rubo) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:74]: I also think we can hardcode this for now although the dictionary gives us the flexibility to add other libs as well. But for now, looks like we don't need that kind of flexibility.

**Review by LukaszRozmej — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**